Source/Engine/Level/Actor.cs

```diff
@@ -116,6 +116,15 @@ public void ResetLocalTransform()
             LocalTransform = Transform.Identity;
         }
 
+        /// <summary>
+        /// Called in-editor when an actor is added to the scene.
+        /// If not in the editor, this function will not be called.
+        /// </summary>
+        public virtual void OnActorSpawned()
```

Comment:
We have `ActorNode` system which handles editor-interactions with actors and scene objects.

Reply:
Alright, I removed `OnActorSpawned` and used BoxColliderNode for this. I did keep the new `BoxCollider.cs` however, as I think that's the best place for the custom editor which adds to manual resizing option.